ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml

```diff
@@ -153,18 +154,23 @@ DappsComboBox {
                 target: root.wcService.requestHandler
 
                 function onMaxFeesUpdated(maxFees, maxFeesWei, haveEnoughFunds, symbol) {
-                    maxFeesText = `${maxFees.toFixed(2)} ${symbol}`
+                    // TODO #15192: handle the case when maxFees is null
+                    dappRequestModal.maxFeesText = `${maxFees.toFixed(2)} ${symbol}`
                     var ethStr = "?"
                     try {
                         ethStr = globalUtils.wei2Eth(maxFeesWei, 9)
                     } catch (e) {
                         // ignore error in case of tests and storybook where we don't have access to globalUtils
                     }
-                    maxFeesEthText = `${ethStr} ETH`
-                    enoughFunds = haveEnoughFunds
+                    dappRequestModal.maxFeesEthText = `${ethStr} ETH`
+                    dappRequestModal.enoughFunds = haveEnoughFunds
                 }
-                function onEstimatedTimeUpdated(minMinutes, maxMinutes) {
-                    estimatedTimeText = qsTr("%1-%2mins").arg(minMinutes).arg(maxMinutes)
+                function onEstimatedTimeUpdated(intervalInMinutes) {
+                    if (!intervalInMinutes) {
+                        dappRequestModal.estimatedTimeText = ""
+                        return
+                    }
+                    dappRequestModal.estimatedTimeText = qsTr("%1mins").arg(intervalInMinutes)
```

Comment:
This in theory needs a proper plural but since Qt works with just one number... you must fully spell out the (known) options. AFAIK we have a function for that somewhere; WalletUtils?

Reply:
Yes, `Constants.TransactionEstimatedTime.XXX` and ` WalletUtils.getLabelForEstimatedTxTime(estimatedTimeAsEnumInt)`

Reply:
Thanks for the heads up, will use them